### PR TITLE
Log nil-coerced uuids in SIP/Prefill forms

### PR DESCRIPTION
**Pull request by markolson**
We've had a few errors trickle in where the non-nil constraint on user_uuid was being hit, which looks like an AR coercion issue. This logs extra data when we hit that case so we can investigate more.

Code in `ensure_uuid` is 1:1 from ActiveRecord to duplicate behavior.

### CODE REVIEW

File: app/controllers/v0/in_progress_forms_controller.rb [hunk 1]
```diff
@@ -29,5 +31,19 @@ def destroy
...
+
+    private
+
+    def ensure_uuid
+      # There have been several errors where `@current_user.uuid` is being coerced to `nil`
+      # by activerecord. This checks the the `uuid` against the same regex and logs an error
+      # if we see a 'malformed' id.
+      uuid_format = ActiveRecord::ConnectionAdapters::PostgreSQL::OID::Uuid::ACCEPTABLE_UUID
+      # We should always have a uuid on @current_user. If this fails, that's another issue
+      unless @current_user.uuid.to_s[uuid_format, 0]
```
Comment (omgitsbillryan) [inline app/controllers/v0/in_progress_forms_controller.rb]: Probably just me but the syntax is a little foreign - is this just performing a [String.match](https://ruby-doc.org/core-2.3.0/String.html#method-i-match)?

File: app/controllers/v0/in_progress_forms_controller.rb [hunk 2]
```diff
@@ -29,5 +31,19 @@ def destroy
...
       render json: form
     end
+
+    private
+
+    def ensure_uuid
+      # There have been several errors where `@current_user.uuid` is being coerced to `nil`
+      # by activerecord. This checks the the `uuid` against the same regex and logs an error
+      # if we see a 'malformed' id.
+      uuid_format = ActiveRecord::ConnectionAdapters::PostgreSQL::OID::Uuid::ACCEPTABLE_UUID
```
Comment (omgitsbillryan) [inline app/controllers/v0/in_progress_forms_controller.rb]: I just looked this up and it takes the form:
```
%r{\A\{?([a-fA-F0-9]{4}-?){8}\}?\z}x
```
My regex is weak - does the `{4}-?` part mean it is supposed to contain 4 dashes?  The UUID in question never contains dashes.

File: app/controllers/v0/in_progress_forms_controller.rb (hunk 1 above)
Comment (markolson) [inline app/controllers/v0/in_progress_forms_controller.rb]: Just copying the AR code verbatim so I know I'm getting exactly the same behavior

**Comment by markolson** [comment]
Re-opening since SiP is going out again

**Pull request merged**